src/mut.js

```diff
@@ -16,13 +17,11 @@ import { updateNestedStateForProperty } from './helpers';
 //     onChange={mut('property', parseFloat)}
 export function mutCreator(context) {
   // mut => the function called by whatever event
-  return (property, transform) => e => {
-    const isFunction = typeof transform === 'function';
+  return (property, opts) => e => {
     const passedValue = findValue(e);
-    const value = isFunction ? transform(passedValue) : passedValue;
 
-    updateNestedStateForProperty(property, value, context);
+    const updateState = value => updateNestedStateForProperty(property, value, context);
 
-    return value;
+    return doLifeCycle(updateState, passedValue, opts);
```

Comment:
so, this method should still exist when you convert the thing to a class.

doLifeCycle would just create the class and run it.